code/modules/vehicles/armored/armored.dm

```diff
@@ -0,0 +1,728 @@
+		var/time = 6 SECONDS - (1 SECONDS * user.skills.getRating("large_vehicle"))
+		if(!do_after(user, time, TRUE, src, BUSY_ICON_BUILD))
+			return
+		if(!get_door_location(user))//in case the tank moves
+			return
+		exit_tank(L)
+		L.SpinAnimation(1,1)
+		L.throw_at(user, 3)
+		// Applying STUN, WEAKEN and STUTTER
+		L.apply_effects(6, 6, 0, 0, 6, 0, 0, 0, 0)
+		user.visible_message("<span class='warning'>[user] rips [L] out of [src] and bodyslams them!.</span>",
+		"<span class='notice'>You rip [L] out of [src] and bodyslam them!.</span>")
+		playsound(get_turf(src), 'sound/effects/throw.ogg', 100, 1)
+		return
+
+	// Entering the tank
+	var/position = alert("What seat would you like to enter?", name, POSITION_DRIVER, POSITION_GUNNER, POSITION_PASSENGER, "Cancel")
+	if(!position || position == "Cancel")
+		return
+	if(pilot && position == POSITION_DRIVER)
+		to_chat(user, "[pilot] is already piloting [src]!")
+		return
+	if(gunner && position == POSITION_GUNNER)
+		to_chat(user, "[gunner] is already gunning [src]!")
+		return
+	if(LAZYLEN(passengers) >= max_passengers && position == POSITION_PASSENGER) //We have a few slots for people to enter as passengers without gunning / driving.
+		to_chat(user, "[src] is full! There isn't enough space for you")
+		return
+	if(!can_enter(user, position)) //OWO can they enter us????	//what the fuck kmc
+		return
+	to_chat(user, "You climb into [src] as a [position]!")
+	enter(user, position)
+
+/obj/vehicle/armored/proc/can_enter(mob/living/carbon/M, position) //NO BENOS ALLOWED
+	if(!istype(M))
+		return
+	if(!M.dextrous)
+		to_chat(M, "<span class='warning'>You don't have the dexterity to drive [src]!</span>")
+		return FALSE
+	if(!open && !allowed(M) && position != POSITION_PASSENGER)
+		to_chat(M, "<span class='warning'>Access denied.</span>")
+		return FALSE
+	var/obj/item/offhand = M.get_inactive_held_item()
+	if(offhand && !(offhand.flags_item & (NODROP|DELONDROP)))
+		to_chat(M, "<span class='warning'>You need your hands free to climb into [src].</span>")
+		return FALSE
+	if(M.skills?.getRating("large_vehicle") < SKILL_LARGE_VEHICLE_TRAINED)
+		M.visible_message("<span class='notice'>[M] fumbles around figuring out how to get into the [src].</span>",
+		"<span class='notice'>You fumble around figuring out how to get into [src].</span>")
+	var/time = 10 SECONDS - (2 SECONDS * M.skills.getRating("large_vehicle"))
+	if(!do_after(M, time, TRUE, src, BUSY_ICON_BUILD))
+		return FALSE
+	if(offhand && !(offhand.flags_item & (NODROP|DELONDROP)))
+		to_chat(M, "<span class='warning'>You need your hands free to climb into [src].</span>")
+		return FALSE
+	if(!Adjacent(M))
+		return FALSE//in case the tank moves
+	return TRUE
+
+
+///attempt to enter the vehicle with a mob
+/obj/vehicle/armored/proc/enter(mob/user, position) //By this point, we've checked that the seats are actually empty, so we won't need to do that again HOPEFULLY
```

Comment:
maybe `enter_tank` instead? which matches the `exit_tank` below.

Reply:
done